src/app/ui/TaskBoard.tsx

```diff
@@ -119,7 +119,7 @@ export const TaskBoard = ({ mode }: TaskBoardProps) => {
           await updateViewModeSettings(z.string().parse(token), payload)
         }}
       />
-      {isNoTasksWithFilter && <NoFilteredTasksState />}
+      {tasks && tasks.length === 0 && <NoFilteredTasksState />}
```

Comment:
When I change a filter like assignee doesn't filteredTasks change instead of tasks? So checking for `tasks.length === 0` would not be optimal.